lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ITaskState.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Interface for task state
+    /// CLient can implement this interface and inject it into context service and task function to save the current task sate
+    /// </summary>
+    [DefaultImplementation(typeof(DefaultTaskState))]
```

Comment:
TaskStateService(ITaskState state) is bound as part of the IMRU service config regardless if user binds the implementation of ITaskState. We need a default implementation in case user doesn't bind his own ITaskState otherwise we will get injection error.